### DISCUSSION TITLE
set name of sampler with specific field value in response

### DISCUSSION THREAD

**Discussion by mehrdad2000** [tilln/jmeter-iso8583#94] [General]
Hi
I want to set name of sampler with specific field value in response, as i debug variable filled with correct value but not show in result tree.

e.g. i create iso8583 sampler to send sample 800 message, after receive response need to put field 2 in response as sampler name.
I use post Boundary Extractor, and it extract variable correctly in debug log but not show in sampler name.

Here is current sampler name: Debit ${NEWPAN}

${NEWPAN} is variable that exist in response of 800 message.

Any idea?
Thanks

**Comment by tilln** [comment]
> ... debug variable filled with correct value but not show in result tree.

I'm afraid I don't understand what your problem is.
If you want to debug, why don't you just write the value to the JMeter log?

> I want to set name of sampler with specific field value in response

I don't think it's a good idea to change the Sampler name as you may want to group results by name later on.

Also you may want to change the name of the SampleResult the Sampler produces instead of the name of the Sampler itself, for example in a JSR223 PostProcessor:
`prev.setSampleLabel(vars.get('NEWPAN'))`

**Reply by mehrdad2000** [reply]
@tilln Finally use post processor to store sampler name and append pan at the end of it. Main purpose of this is to store summary of output report without headache.  
prev.setSampleLabel(prev.getSampleLabel() + " " + vars.get("NEWPAN"))

Thanks
